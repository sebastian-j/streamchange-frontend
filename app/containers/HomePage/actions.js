/*
 *
 * HomePage actions
 *
 */

import {
  CHANGE_OWNER_ID,
  CHANGE_THUMBNAIL_URL,
  CHANGE_TITLE,
  CHANGE_VIDEO_ID,
} from './constants';

export function changeOwnerId(id) {
  return {
    type: CHANGE_OWNER_ID,
    ownerId: id,
  };
}

export function changeThumbnailUrl(url) {
  return {
    type: CHANGE_THUMBNAIL_URL,
    thumbnailUrl: url,
  };
}

export function changeTitle(t) {
  return {
    type: CHANGE_TITLE,
    title: t,
  };
}

export function changeVideoId(id) {
  return {
    type: CHANGE_VIDEO_ID,
    videoId: id,
  };
}
