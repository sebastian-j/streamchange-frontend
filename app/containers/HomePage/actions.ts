/*
 *
 * HomePage actions
 *
 */
import { action } from 'typesafe-actions';
import ActionTypes from './constants';
import { Ban, Stream } from './types';

export const changeStreamProperties = (stream: Stream) =>
  action(ActionTypes.CHANGE_STREAM_PROPERTIES, stream);

export const changeBanStatus = (ban: Ban | null) =>
  action(ActionTypes.CHANGE_BAN_STATUS, ban);

export const loadAuthKey = (authKey: string) =>
  action(ActionTypes.LOAD_AUTH_KEY, authKey);

export const sendTelemetryData = (stream: Stream) =>
  action(ActionTypes.SEND_TELEMETRY_DATA, stream);
