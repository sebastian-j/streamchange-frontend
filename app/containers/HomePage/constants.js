/*
 *
 * HomePage constants
 *
 */

export const CHANGE_OWNER_ID = 'app/HomePage/CHANGE_OWNER_ID';
export const CHANGE_THUMBNAIL_URL = 'app/HomePage/CHANGE_THUMBNAIL_URL';
export const CHANGE_TITLE = 'app/HomePage/CHANGE_THUMBNAIL_URL';
export const CHANGE_VIDEO_ID = 'app/HomePage/CHANGE_THUMBNAIL_URL';
