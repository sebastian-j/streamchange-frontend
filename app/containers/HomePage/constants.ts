/*
 *
 * HomePage constants
 *
 */
enum ActionTypes {
  CHANGE_OWNER_ID = 'app/HomePage/CHANGE_OWNER_ID',
  CHANGE_THUMBNAIL_URL = 'app/HomePage/CHANGE_THUMBNAIL_URL',
  CHANGE_TITLE = 'app/HomePage/CHANGE_THUMBNAIL_URL',
  CHANGE_VIDEO_ID = 'app/HomePage/CHANGE_THUMBNAIL_URL',
}

export default ActionTypes;
