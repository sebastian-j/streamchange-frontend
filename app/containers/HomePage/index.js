/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import axios from 'axios';
import Button from '@material-ui/core/Button';
import WelcomeDialog from '../../components/WelcomeDialog';
import YoutubeWorker from '../../components/YoutubeWorker';
import './style.css';

export default class HomePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      channelId: '',
      videoId: '',
      title: '',
      thumbnailUrl: '',
      liveChatId: '',
    };
    this.receiveVideo = this.receiveVideo.bind(this);
    this.leaveStream = this.leaveStream.bind(this);
    this.launchWorker = this.launchWorker.bind(this);
  }

  receiveVideo(videoLink) {
    if (videoLink.includes('v=')) {
      let videoId = videoLink.split('v=')[1];
      videoId = videoId.split('&')[0];
      videoId = videoId.split('/')[0];
      this.launchWorker(videoId);
    }
  }

  leaveStream() {
    this.setState({ videoId: '', title: '', liveChatId: '', thumbnailUrl: '' });
    sessionStorage.removeItem('gv-videoId');
    window.location.reload();
  }

  launchWorker(videoId) {
    axios
      .get(
        `https://www.googleapis.com/youtube/v3/videos?part=snippet%2C+liveStreamingDetails&id=${videoId}&key=API_KEY`,
      )
      .then(res => {
        this.setState({
          videoId,
          channelId: res.data.items[0].snippet.channelId,
          title: res.data.items[0].snippet.title,
          thumbnailUrl: res.data.items[0].snippet.thumbnails.medium.url,
          liveChatId: res.data.items[0].liveStreamingDetails.activeLiveChatId,
        });
        sessionStorage.setItem('gv-videoId', videoId);
      });
  }

  componentDidMount() {
    const id = sessionStorage.getItem('gv-videoId');
    if (id !== null) {
      this.launchWorker(id);
    }
  }

  render() {
    if (this.state.videoId === '') {
      return <WelcomeDialog passVideo={this.receiveVideo} />;
    }
    return (
      <div>
        <div className="stream-info">
          <img alt="Miniatura" src={this.state.thumbnailUrl} />
          <span>{this.state.title}</span>
          <Button onClick={this.leaveStream} color="primary">
            Opuść stream
          </Button>
        </div>
        <YoutubeWorker
          channelId={this.state.channelId}
          liveChatId={this.state.liveChatId}
          videoId={this.state.videoId}
          apiKey="API_KEY"
        />
      </div>
    );
  }
}
