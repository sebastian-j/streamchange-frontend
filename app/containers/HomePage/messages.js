/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'streamchange.containers.HomePage';

export default defineMessages({
  historyLink: {
    id: `${scope}.historyLink`,
    defaultMessage: 'Win history',
  },
  leaveStreamBtn: {
    id: `${scope}.leaveStreamBtn`,
    defaultMessage: 'Leave stream',
  },
});
