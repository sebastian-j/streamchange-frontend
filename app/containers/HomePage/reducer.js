/*
 *
 * HomePage reducer
 *
 */
import produce from 'immer';

import {
  CHANGE_OWNER_ID,
  CHANGE_THUMBNAIL_URL,
  CHANGE_TITLE,
  CHANGE_VIDEO_ID,
} from './constants';

export const initialState = {
  ownerId: '',
  thumbnailUrl: '',
  title: '',
  videoId: '',
};

/* eslint-disable default-case, no-param-reassign */
const homePageReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_OWNER_ID:
        draft.ownerId = action.ownerId;
        break;
      case CHANGE_THUMBNAIL_URL:
        draft.thumbnailUrl = action.thumbnailUrl;
        break;
      case CHANGE_TITLE:
        draft.title = action.title;
        break;
      case CHANGE_VIDEO_ID:
        draft.videoId = action.videoId;
        break;
    }
  });

export default homePageReducer;
