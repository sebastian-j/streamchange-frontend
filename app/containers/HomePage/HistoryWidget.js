import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { withStyles } from '@material-ui/core/styles';
import Tooltip from '@material-ui/core/Tooltip';

import messages from './messages';
import db from '../../components/YoutubeWorker/db';

const StyledLink = styled.span`
  background: ${(props) => props.theme.iconButtonBackground};
  border: none;
  border-radius: 6px;
  color: ${(props) => props.theme.buttonTextColor};
  font-size: 1.05rem;
  height: 80%;
  padding: 8px 8px;
  margin: 0 15px 0 0;
  text-decoration: none;
  &:hover {
    background-color: ${(props) => props.theme.buttonBackgroundHover};
    color: ${(props) => props.theme.buttonTextColorHover};
  }
  @media (orientation: portrait) {
    line-height: 3em;
  }
`;

const WinnerTableTitle = styled.div`
  font-size: 14px;
  margin: 5px 0;
  text-align: center;
`;

const WarningContent = styled.div`
  font-size: 13px;
`;
const Td = styled.td`
  padding-right: 5px;
  vertical-align: middle;
`;

const ExtendedTooltip = withStyles((theme) => ({
  tooltip: {
    backgroundColor: 'rgba(225,246,246,0.9)',
    color: 'rgba(0, 0, 0, 0.87)',
    maxWidth: '300px',
    fontSize: theme.typography.pxToRem(12),
    border: '1px solid #949499',
  },
}))(Tooltip);

const HistoryWidget = () => {
  const [winners, setWinners] = useState([]);
  const [warning, setWarning] = useState(false);
  const getHistory = () => {
    db.table('history')
      .reverse()
      .toArray()
      .then((items) => {
        setWarning(items.length > 100);
        const it = Array.from(items);
        if (it.length > 3) it.length = 3;
        setWinners(it);
      })
      .catch(() => {
        setWinners([]);
      });
  };

  useEffect(() => {
    getHistory();
  }, []);

  return (
    <ExtendedTooltip
      title={
        <>
          {!warning && winners.length === 0 && (
            <WarningContent>
              <FormattedMessage {...messages.historyEmptyTooltip} />
            </WarningContent>
          )}
          {warning && (
            <WarningContent>
              <FormattedMessage {...messages.historyWidgetWarningText} />
            </WarningContent>
          )}
          {winners.length > 0 && (
            <WinnerTableTitle>
              <FormattedMessage {...messages.historyWidgetTableTitle} />
            </WinnerTableTitle>
          )}
          {winners.map((item) => (
            <tr key={item.createdAt}>
              <Td>
                <img src={item.imageUrl} alt="Logo" width="32px" />
              </Td>
              <Td>{item.displayName}</Td>
              <Td>{item.prize}</Td>
            </tr>
          ))}
        </>
      }
    >
      <NavLink to="/giveaway-history" style={{ textDecoration: 'none' }}>
        <StyledLink onMouseEnter={getHistory}>
          <FormattedMessage {...messages.historyLink} />
          {warning && (
            <span role="img" aria-label="warning">
              ⚠️
            </span>
          )}
        </StyledLink>
      </NavLink>
    </ExtendedTooltip>
  );
};

export default HistoryWidget;
