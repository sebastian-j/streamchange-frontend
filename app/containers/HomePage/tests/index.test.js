import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import HomePage from '../index';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
