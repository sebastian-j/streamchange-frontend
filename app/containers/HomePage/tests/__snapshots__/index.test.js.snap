// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="FirstUseScreen__Backdrop-sc-18wtu21-0 fBIFTj"
>
  <span
    class="FirstUseScreen__Title-sc-18wtu21-1 dTkyKr"
  >
    <span>
      Welcome to StreamChange
    </span>
  </span>
  <div
    class="tiles-grid"
  >
    <span
      class="FirstUseScreen__Title-sc-18wtu21-1 dTkyKr"
    >
      <span>
        Select theme
      </span>
    </span>
    <label
      for="themeLight"
    >
      <input
        checked=""
        class="FirstUseScreen__RadioInput-sc-18wtu21-2 hVNXhY"
        id="themeLight"
        name="theme"
        type="radio"
        value="0"
      />
      <div
        class="box FirstUseScreen__Box-sc-18wtu21-3 bPAypJ"
      >
        <img
          alt="light"
          class="FirstUseScreen__ThemeImg-sc-18wtu21-8 lmaBwM"
          src="IMAGE_MOCK"
        />
        <span
          class="FirstUseScreen__Tile-sc-18wtu21-5 kyBueP"
        >
          <span>
            Light
          </span>
        </span>
      </div>
    </label>
    <label
      for="themeDark"
    >
      <input
        class="FirstUseScreen__RadioInput-sc-18wtu21-2 hVNXhY"
        id="themeDark"
        name="theme"
        type="radio"
        value="1"
      />
      <div
        class="box FirstUseScreen__Box-sc-18wtu21-3 kcBKzb"
      >
        <img
          alt="dark"
          class="FirstUseScreen__ThemeImg-sc-18wtu21-8 lmaBwM"
          src="IMAGE_MOCK"
        />
        <span
          class="FirstUseScreen__Tile-sc-18wtu21-5 kyBueP"
        >
          <span>
            Dark
          </span>
        </span>
      </div>
    </label>
  </div>
  <div
    class="tiles-grid"
  >
    <span
      class="FirstUseScreen__Title-sc-18wtu21-1 dTkyKr"
    >
      <span>
        Select language
      </span>
    </span>
    <label
      for="pl"
    >
      <input
        class="FirstUseScreen__RadioInput-sc-18wtu21-2 hVNXhY"
        id="pl"
        name="language"
        type="radio"
        value="pl"
      />
      <div
        class="box FirstUseScreen__LangBox-sc-18wtu21-4 iFLZEI"
      >
        <span
          class="FirstUseScreen__LangTile-sc-18wtu21-6 gDeyJE"
        >
          <img
            alt="pl"
            class="FirstUseScreen__Flag-sc-18wtu21-7 fyboRz"
            src="IMAGE_MOCK"
          />
          <span>
            pl
          </span>
        </span>
      </div>
    </label>
    <label
      for="en"
    >
      <input
        checked=""
        class="FirstUseScreen__RadioInput-sc-18wtu21-2 hVNXhY"
        id="en"
        name="language"
        type="radio"
        value="en"
      />
      <div
        class="box FirstUseScreen__LangBox-sc-18wtu21-4 hsnTTl"
      >
        <span
          class="FirstUseScreen__LangTile-sc-18wtu21-6 gDeyJE"
        >
          <img
            alt="en"
            class="FirstUseScreen__Flag-sc-18wtu21-7 fyboRz"
            src="IMAGE_MOCK"
          />
          <span>
            en
          </span>
        </span>
      </div>
    </label>
  </div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text FirstUseScreen__StyledButton-sc-18wtu21-9 eyFmjv"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span>
          Save
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
